tests/trx_generator/trx_provider.hpp

```diff
@@ -55,8 +58,28 @@ namespace eosio::testing {
          _connection_thread_pool.stop();
       };
 
+      fc::time_point get_trx_ack_time(const eosio::chain::transaction_id_type& _trx_id) {
+         fc::time_point time_acked;
+         _trx_ack_map_lock.lock();
+         auto search = _trxs_ack_time_map.find(_trx_id);
+         if (search != _trxs_ack_time_map.end()) {
+            time_acked = search->second;
+         } else {
+            elog("get_trx_ack_time - Transaction acknowledge time not found for transaction with id: ${id}", ("id", _trx_id));
+            time_acked = fc::time_point::min();
+         }
+         _trx_ack_map_lock.unlock();
+         return time_acked;
+      }
+
       virtual void send_transaction(const chain::packed_transaction& trx) = 0;
 
+      void trx_acknowledged(const eosio::chain::transaction_id_type _trx_id, const fc::time_point ack_time) {
+         _trx_ack_map_lock.lock();
```

Comment:
Use a `std::lock_guard g(_trx_ack_map_lock);`

Reply:
Thanks. Had been planning to do this, but completely forgot to come back and change it.
Addressed: [400c5b8](https://github.com/AntelopeIO/leap/pull/1165/commits/400c5b891a9f12e86916b246d1e4a073e39ec9e0)